### ISSUE TITLE
When running wasm-bindgen-cli with a crate compiled on the wrong version of wasm-bindgen, we should error

### ISSUE THREAD

**Issue by pranaygp** [rustwasm/wasm-bindgen#235]
I ran through the example in the README and everything worked perfectly. Then, I attempted at setting up [this repo](https://github.com/atom/xray) locally. Specifically, look at the https://github.com/atom/xray/tree/master/xray_wasm project.

I'm unable to specifically figure out where everything is going wrong, but I began to debug why running the existing tests were failing (~they work on CI, just not when I run it locally~ on CI and locally) and it seems like calls to JS functions from rust aren't passed the argument. Specifically, this call (https://github.com/atom/xray/blob/master/xray_wasm/src/lib..rs#L237) which sends an `item`. Further, I also created an additional function in JS that simply takes an argument and `console.log`s it, added it to the `extern` definition in the rust file, and called it at multiple places with direct strings like `myLogFn("ABC")`. In this case, my JS function gets called correctly, except the argument which is ALWAYS set to "" (empty string).

So I began to dig into this further and, in the output `xray_wasm.js` file, I found the `__wbg_f_log_log2_n` function definition, and modified it slightly to log the incoming arguments:

```javascript
export function __wbg_f_log_log2_n(arg0, arg1) {
    console.log(arguments); // This is the only line I added
    let varg0 = getStringFromWasm(arg0, arg1);
    log(varg0);
}
```

I then build and run my code using these commands (how the project is supposed to be [run](https://github.com/atom/xray/blob/master/xray_wasm/script/test)):
```bash
node_modules/.bin/webpack --target=node --mode=development --devtool="source-map" test/tests.js
node_modules/.bin/mocha --require source-map-support/register --ui=tdd dist/main.js
```

You'll notice that the output from every console.log is:
```javascript
{ '0': 4140 }
{ '0': 4151 }
{ '0': 4151 }
{ '0': 4151 }
...
```

So here, the empty string arguments completely make sense because the call to `getStringFromWasm` is being passed in the correct argument for the starting memory location, but the length is undefined. I confirmed this by modifying `getStringFromWasm` to ignore the `len` param and just use an arbitrary number like `3`, and you'll see the correct output ("ABC"). 

This is as far as I could drill down because the actual callsite for `__wbg_f_log_log2_n` is in `wasm` (am I wrong?) and so I can't figure out why it's not being called correctly with the length param. ¯\\_(ツ)_/¯

**Comment by sendilkumarn** [comment]
Can you update the wasm-bindgen-cli and try?

**Comment by pranaygp** [comment]
Just did, the issue remains: FYI, I'm on macOS High Sierra and my wasm-bindgen-cli is on v0.2.11

**Comment by sendilkumarn** [comment]
Yeah, I also faced this issue when the versions mismatch. 

@alexcrichton Can we check on the versions to highlight users? WDYT?

**Comment by sendilkumarn** [comment]
Okay I will spend sometime in the weekend 👍

**Comment by alexcrichton** [comment]
Support for this error comes from this block of code -- 

https://github.com/rustwasm/wasm-bindgen/blob/2f465d3d2329055bfd16148fa2fa095cfbe21688/crates/cli-support/src/lib.rs#L252-L275

Empirically that doesn't seem to be working out much in practice, and I think that it's due to the fact that the "schema version" written down there must be hand-modified, and I often forget to do so when making a release. The schema needs to be modified basically whenever the ABI or format of the custom section changes. I believe that in theory if the schema version is updated regularly and correctly this problem would be solved (you'd get an error on a mismatch).

It's also worth noting that historically I believe the schema version check was slightly incorrect in that it accepted or rejected too much, which also has historically reduced its usefulness.

I tried initially to just set the schema version to the version of the crate as well (which is always updated whenever a release is done), but I found that it was a bit of a pain to always reinstall wasm-bindgen when not all point releases needed it. It may be worthwhile to go back to that though if we don't have a lot of bugfix releases?
